### ISSUE TITLE
doesn't work on different interfaces

### ISSUE THREAD

**Issue by lumpov** [gr3p1p3/transparent-proxy#2]
Clear raspberry OS install.

```
pi@raspberrypi:~ $ uname -a
Linux raspberrypi 5.10.17+ #1403 Mon Feb 22 11:26:13 GMT 2021 armv6l GNU/Linux
```

```
pi@raspberrypi:~ $ ifconfig
eth0: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        ether b8:27:eb:25:06:5b  txqueuelen 1000  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

eth1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.8.100  netmask 255.255.255.0  broadcast 192.168.8.255
        inet6 fe80::442f:6c50:3cbf:3552  prefixlen 64  scopeid 0x20<link>
        ether 0c:5b:8f:27:9a:64  txqueuelen 1000  (Ethernet)
        RX packets 104  bytes 16704 (16.3 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 197  bytes 15659 (15.2 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 2  bytes 78 (78.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 2  bytes 78 (78.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

wlan0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.88.237  netmask 255.255.255.0  broadcast 192.168.88.255
        inet6 fe80::ca41:6ad9:26ff:a3a0  prefixlen 64  scopeid 0x20<link>
        ether 00:13:ef:70:05:a9  txqueuelen 1000  (Ethernet)
        RX packets 5067  bytes 771140 (753.0 KiB)
        RX errors 0  dropped 1162  overruns 0  frame 0
        TX packets 198  bytes 36153 (35.3 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```

```
pi@raspberrypi:~/Development/proxy $ route
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
default         hi.link         0.0.0.0         UG    203    0        0 eth1
default         192.168.88.1    0.0.0.0         UG    204    0        0 wlan0
192.168.8.0     0.0.0.0         255.255.255.0   U     203    0        0 eth1
192.168.88.0    0.0.0.0         255.255.255.0   U     204    0        0 wlan0
```

```
const ProxyServer = require('transparent-proxy');

const server = new ProxyServer({
    verbose: true,
    tcpOutgoingAddress: function (data, bridgedConnection) {
//        return '192.168.8.100';
	return '192.168.88.237';
    },
});

server.listen(8080, '192.168.88.237', function () {
    console.log('TCP-Proxy-Server started!', server.address());
});
```

Start the server and receive request:
```
pi@raspberrypi:~/Development/proxy $ node server
TCP-Proxy-Server started! { address: '192.168.88.237', family: 'IPv4', port: 8080 }
### 2021-03-25T07:59:50.342Z 192.168.88.249:52132 => { ADDRESS: 'ifconfig.co', PORT: 443 }
```

Request from local network:
```
lumpov@workgroup kassa % curl -x 192.168.88.237:8080 https://ifconfig.co
curl: (56) Proxy CONNECT aborted
```

But if I try to use different interfaces - it is ok:
```
pi@raspberrypi:~ $ curl --interface wlan0 ifconfig.co
58.213.133.126
```

```
pi@raspberrypi:~ $ curl --interface eth1 ifconfig.co
85.26.165.122
```

**Comment by gr3p1p3** [comment]
Fixed, please download @1.6.4 :) and let me know if it works!

**Comment by gr3p1p3** [comment]
Ok I setup a server to test this issue :)

```bash
root@vps73857815194:/home/proxyTest# uname -a
Linux vps73857815194 5.4.103-1-pve #1 SMP PVE 5.4.103-1 (Sun, 07 Mar 2021 15:55:09 +0100) x86_64 x86_64 x86_64 GNU/Linux
root@vps73857815194:/home/proxyTest# ifconfig
eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 5.230.70.118  netmask 255.255.255.0  broadcast 5.230.70.255
        inet6 fe80::806e:5dff:febb:bdee  prefixlen 64  scopeid 0x20<link>
        ether 82:6e:5d:bb:bd:ee  txqueuelen 1000  (Ethernet)
        RX packets 2257060  bytes 688355452 (688.3 MB)
        RX errors 0  dropped 239428  overruns 0  frame 0
        TX packets 233  bytes 29200 (29.2 KB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

eth1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 5.230.68.16  netmask 255.255.255.0  broadcast 5.230.68.255
        inet6 fe80::f0b0:afff:fe01:7cc7  prefixlen 64  scopeid 0x20<link>
        ether f2:b0:af:01:7c:c7  txqueuelen 1000  (Ethernet)
        RX packets 2285902  bytes 842253206 (842.2 MB)
        RX errors 0  dropped 239242  overruns 0  frame 0
        TX packets 23065  bytes 2658019 (2.6 MB)
        TX errors 0  dropped 1 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 257  bytes 59283 (59.2 KB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 257  bytes 59283 (59.2 KB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

```

```javascript
const ProxyServer = require('transparent-proxy');

const server = new ProxyServer({
    verbose: true,
    tcpOutgoingAddress: function (data, bridgedConnection) {
        //return '5.230.70.118';
        return '5.230.68.16';
    },
});

server.listen(8080,'5.230.70.118',function () {console.log('TCP-Proxy-Server started!', server.address());});
```

```bash
node test.js
TCP-Proxy-Server started! { address: '5.230.70.118', family: 'IPv4', port: 8080 }
```

```bash
root@vps73857815194:/home/proxyTest# curl -x 5.230.70.118:8080 https://ifconfig.me/
5.230.68.16
```
